judge/models/problem.py

```diff
@@ -535,3 +558,37 @@ class Meta:
         )
         verbose_name = _('solution')
         verbose_name_plural = _('solutions')
+
+
+class ProblemPointsVote(models.Model):
+    points = models.IntegerField(
+        verbose_name=_('proposed points'),
+        help_text=_('The amount of points the voter thinks this problem deserves.'),
+        validators=[
+            MinValueValidator(settings.DMOJ_PROBLEM_MIN_USER_POINTS_VOTE),
+            MaxValueValidator(settings.DMOJ_PROBLEM_MAX_USER_POINTS_VOTE),
+        ],
+    )
+
+    voter = models.ForeignKey(Profile, related_name='problem_points_votes', on_delete=CASCADE)
+    problem = models.ForeignKey(Problem, related_name='problem_points_votes', on_delete=CASCADE)
+    vote_time = models.DateTimeField(
+        verbose_name=_('vote time'),
+        help_text=_('The time this vote was cast.'),
+        auto_now_add=True,
+    )
+
+    note = models.TextField(
+        verbose_name=_('note'),
+        help_text=_('Justification for problem point value.'),
+        max_length=2048,
+        blank=True,
+        default='',
+    )
+
+    class Meta:
+        verbose_name = _('problem vote')
+        verbose_name_plural = _('problem votes')
+
+    def __str__(self):
+        return f'Points vote by {self.voter} for {self.problem}'
```

Comment:
```suggestion
        return _('Points vote by %(voter)s for %(problem)s') % {'voter': self.voter, 'problem': self.problem}
```
Following https://github.com/DMOJ/online-judge/pull/1932